jdbc-yugabytedb/src/main/java/org/postgresql/Driver.java

```diff
@@ -477,17 +477,18 @@ private static Connection getConnectionBalanced(LoadBalanceProperties lbprops) t
     Set<String> unreachableHosts = loadBalancer.getUnreachableHosts();
     List<String> failedHosts = new ArrayList<>(unreachableHosts);
     String chosenHost = loadBalancer.getLeastLoadedServer(failedHosts);
-    Connection newConnection = null;
-    Connection controlConnection = null;
+    PgConnection newConnection = null;
```

Comment:
What's the difference between `Connection` and `PgConnection` object?

Reply:
It is the PG implementation of the Connection interface of JDBC. I wanted to avoid an unnecessary cast at this level so using the impl class directly.